sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/ODataDataObject.scala

```diff
@@ -0,0 +1,644 @@
+package io.smartdatalake.workflow.dataobject
+
+import com.typesafe.config.Config
+import io.smartdatalake.config.SdlConfigObject.DataObjectId
+import io.smartdatalake.config.{ConfigurationException, FromConfigFactory, InstanceRegistry}
+import io.smartdatalake.util.secrets.StringOrSecret
+import io.smartdatalake.definitions.OAuthMode
+
+import scala.collection.mutable.ArrayBuffer
+import io.smartdatalake.util.hdfs.{HdfsUtil, PartitionValues}
+import io.smartdatalake.util.misc.SmartDataLakeLogger
+import io.smartdatalake.util.webservice.WebserviceMethod.WebserviceMethod
+import io.smartdatalake.util.webservice.{ScalaJWebserviceClient, WebserviceException, WebserviceMethod}
+import io.smartdatalake.workflow.action.executionMode.DataObjectStateIncrementalMode
+import io.smartdatalake.workflow.dataframe.GenericSchema
+import io.smartdatalake.workflow.dataframe.spark.{SparkSchema, SparkSubFeed}
+import io.smartdatalake.workflow.{ActionPipelineContext, ExecutionPhase}
+import org.apache.hadoop.fs.FileSystem
+import org.apache.spark.sql.{DataFrame, functions}
+import org.apache.spark.sql.types.{ArrayType, DateType, StringType, StructField, StructType, TimestampType}
+import org.apache.spark.sql.functions.{col, date_format, expr, max}
+import org.json4s.jackson.Serialization
+import org.json4s.{DefaultFormats, Formats}
+
+import java.net.URLEncoder
+import java.nio.file.{Files, Paths}
+import java.io.{BufferedWriter, File, FileWriter}
+import java.time.{Instant, ZoneId}
+import scala.annotation.tailrec
+import scala.util.{Failure, Success}
+import org.apache.hadoop.fs.{Path => HadoopPath}
+import org.apache.spark.sql.custom.ExpressionEvaluator
+
+import scala.reflect.runtime.universe.typeOf
+
+/**
+ * InversionOfControl container class
+ * This class serves as a hub or a relay for all calls to class creators and to static functions outside of this module.
+ * It allows test code to override these calls and to inject Mock- and Spy objects into the running code.
+ */
+class ODataIOC {
```

Comment:
As this is state less, it could be an "object" instead of "class".

Reply:
Converting to an object may lead the problems during the UnitTests. The purpose of this class is to use the Mockito Spy feature to override methods and inject test objects like Mocks into the execution.